feat: Add support for Next.js 16 in peerDependencies

aularon opened the pull request.

aularon committed:
Add support for Next.js 16 in peerDependencies

amannn commented:
Thanks for the PR! Can you update the examples as well so we can test if this works without changes?

aularon committed:
Fix linting issue

aularon committed:
Update `example-app` to next 16

aularon commented:
> Thanks for the PR! Can you update the examples as well so we can test if this works without changes?

I updated the app example and fixed a linting issue in the pages one.

Let me know if I should do more updates. Thank you!

aularon commented:
I just figured out that `next-query-params` use is not covered in tests in the project I'm working on. Is the included app example building a signal good enough to make sure all is good with nextjs 16? Otherwise you may want to hold this PR until one is sure enough!

amannn commented on packages/example-app/package.json:
```diff
@@ -10,17 +10,17 @@
-    "react-dom": "^18.2.0",
+    "react": "^19.2.3",
+    "react-dom": "^19.2.3",
     "use-query-params": "2.0.0"
```
While we're at it, would you mind incorporating this?

```suggestion
    "use-query-params": "^2.2.2"
```

amannn commented on packages/example-app/package.json:
```diff
@@ -10,17 +10,17 @@
   },
   "dependencies": {
-    "next": "^13.4.12",
+    "next": "^16.0.10",
```
It seems like the node runner in the GitHub workflow needs to be updated too:

```
Expected version ">=20.9.0". Got "18.20.8"
```

[(ref)](https://github.com/amannn/next-query-params/actions/runs/20303139624/job/58315847339?pr=49)

aularon replied:
Done & done!

amannn reviewed (commented):
> Is the included app example building a signal good enough to make sure all is good with nextjs 16?

That should be fine!

aularon committed:
Use node 24.x in GitHub workflows

aularon committed:
use-query-params@^2.2.2

amannn commented:
Awesome, thanks! It seems like there's an error in one of the tests: https://github.com/amannn/next-query-params/actions/runs/20308332213/job/58332641251?pr=49

Can you have a look?
